config/initializers/prometheus.rb

```diff
@@ -1,18 +1,39 @@
-if Settings.prometheus_exporter_host.present? && !$0.include?('prometheus_exporter')
+unless Rails.env.test?
+  require 'prometheus_exporter'
   require 'prometheus_exporter/client'
   require 'prometheus_exporter/metric'
   require 'prometheus_exporter/middleware'
   require 'prometheus_exporter/instrumentation'
-  client = PrometheusExporter::Client.new(
-    host: Settings.prometheus_exporter_host,
-    port: Settings.prometheus_exporter_port
-    # Add custom_labels for app_name here (consumer and webserver)
-  )
+
+  def ensure_exporter_server
+    require 'socket'
+    TCPSocket.open("127.0.0.1", Settings.metrics_port) {}
+  rescue Errno::ECONNREFUSED
+    require 'prometheus_exporter/server'
+
+    server = PrometheusExporter::Server::WebServer.new(port: Settings.metrics_port)
+    server.start
+
+    if $0.include?('puma') # only one pod to report DB metrics
```

Comment:
I tried `rails`, `puma`, and `rack` here without success so far. I'm testing by deploying in the ephemeral environment, waiting for init containers to complete, then opening a terminal in each pod and running `curl 127.0.0.1:9000/metrics`. @vkrizan maybe you could investigate it as well? I tried adding some log statements to see what `$0` is for each pod, but the logs didn't show any output for some reason (I tried `puts` and `Rails.logger.info`). Otherwise, I'll continue digging tomorrow.